HNC: make target kind-deploy is failing due to empty PROJECT_ID

While deploying HNC in a kind cluster using `make kind-deploy` or `CONFIG=kind make deploy`, the deployment fails with below error.
```
$ make kind-deploy        
CONFIG=kind make deploy
make[1]: Entering directory '/home/mugdha/codespace/mugdha-adhav/multi-tenancy/incubator/hnc'
/home/mugdha/codespace/mugdha-adhav/multi-tenancy/incubator/hnc/bin/controller-gen object:headerFile=./hack/boilerplate.go.txt paths=./api/...
go fmt ./api/... ./cmd/... ./internal/... ./third_party/...
go vet ./api/... ./cmd/... ./internal/... ./third_party/...
Warning: this does not run tests. Run 'make test' to ensure tests are passing.
docker build . -t gcr.io//hnc-manager:kind-local
invalid argument "gcr.io//hnc-manager:kind-local" for "-t, --tag" flag: invalid reference format
See 'docker build --help'.
make[1]: *** [Makefile:212: docker-build] Error 125
make[1]: Leaving directory '/home/mugdha/codespace/mugdha-adhav/multi-tenancy/incubator/hnc'
make: *** [Makefile:230: kind-deploy] Error 2
```

The error occurs as `PROJECT_ID` variable is unset in case of kind deployment. The variable value is retrieved by running gcloud command which is not required for kind deployment.

--- a/incubator/hnc/Makefile
+++ b/incubator/hnc/Makefile
@@ -22,7 +22,9 @@ PROJECT_ID := ${shell gcloud config get-value project 2>/dev/null}
 # By default, we use the Container Registry in the current GCP project, but you
 # can set it to anything, UNLESS you are calling 'make release' since the image
 # needs to be built in the same project as the GCB instance that builds it.
-HNC_REGISTRY ?= gcr.io/${PROJECT_ID}
+ifdef PROJECT_ID
+  HNC_REGISTRY ?= gcr.io/${PROJECT_ID}
+endif
 
 # The image name is the *base* name - excluding the registry and the tag.
 HNC_IMG_NAME ?= hnc-manager
@@ -44,7 +46,11 @@ endif
 #
 # Note that if you're using Kind, this image will never actually be pushed to
 # the registry.
-HNC_IMG = ${HNC_REGISTRY}/${HNC_IMG_NAME}:${HNC_IMG_TAG}
+ifdef HNC_REGISTRY
+	HNC_IMG = ${HNC_REGISTRY}/${HNC_IMG_NAME}:${HNC_IMG_TAG}
+else
+	HNC_IMG = ${HNC_IMG_NAME}:${HNC_IMG_TAG}
+endif
 
 # `make` must be called from the HNC root, or all kinds of things will break
 # (starting with this).

